api/src/routes/discoverable-guilds.ts

```diff
@@ -1,16 +1,19 @@
-import { Guild } from "@fosscord/util";
+import { Guild, Config } from "@fosscord/util";
+
 import { Router, Request, Response } from "express";
 import { route } from "@fosscord/api";
 
+
 const router = Router();
 
 router.get("/", route({}), async (req: Request, res: Response) => {
 	const { limit } = req.params;
+	var showAllGuilds = Config.get().guild.showAllGuildsInDiscovery;
 
 	// ! this only works using SQL querys
 	// TODO: implement this with default typeorm query
 	// const guilds = await Guild.find({ where: { features: "DISCOVERABLE" } }); //, take: Math.abs(Number(limit)) });
-	const guilds = await Guild.find({ where: `"features" LIKE 'COMMUNITY'`, take: Math.abs(Number(limit)) });
+	const guilds = showAllGuilds ? await Guild.find({take: Math.abs(Number(limit))}) : await Guild.find({ where: `"features" LIKE '%COMMUNITY%'`, take: Math.abs(Number(limit)) });
```

Comment:
You need to have a default value for limit if the user doesn't supply the query.